model_analyzer/result/run_config_measurement.py

```diff
@@ -0,0 +1,373 @@
+# Copyright (c) 2022 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from model_analyzer.constants import COMPARISON_SCORE_THRESHOLD
+from model_analyzer.constants import LOGGER_NAME
+
+from model_analyzer.result.model_config_measurement import ModelConfigMeasurement
+from model_analyzer.record.record import RecordType
+
+from statistics import mean
+
+from functools import total_ordering
+import logging
+
+logger = logging.getLogger(LOGGER_NAME)
+
+
+@total_ordering
+class RunConfigMeasurement:
+    """
+    Encapsulates the set of metrics obtained from all model configs
+    in a single RunConfig
+    """
+
+    def __init__(self, key, gpu_data):
+        """
+        key: str
+        Unique string which can be used to differentiate this measurement 
+        from all others in the model
+        
+        gpu_data : dict of list of Records
+            Metrics from the monitors that have a GPU UUID
+            associated with them            
+        """
+        self._key = key
+
+        self._gpu_data = gpu_data
+        self._avg_gpu_data = self._average_list(list(self._gpu_data.values()))
+
+        self._model_config_measurements = []
+        self._model_config_weights = []
```

Comment:
Is there no default? Is it required to set this before doing any operations?  Should it be part of from_dict?

Reply:
The requirement is that this function is called before any comparisons are done. This is not a part of the dict as it is set during the analyze step (from_dict is only relevant for restoring values from the profile step).

The expectation is that model and metric weighting will be set before any comparisons are done.

Reply:
"The expectation is that model and metric weighting will be set before any comparisons are done"

Is there any assert to protect this statement?

Reply:
No assert needed. We will just report that the run's are equivalent. See my change in MCM